### PR TITLE
Automatic failure analysis

### PR DESCRIPTION
**Pull request by PCSwingle**

### CODE REVIEW

**Comment by biobootloader** [comment]
Very nice stuff!

How much does it cost to run on how many exercism results?

Including some screenshots in the PR description of the html results page would be good practice for changes like this, so it's really fast for people to see what it looks like.

File: tests/benchmarks/exercism_practice.py [hunk 1]
```diff
@@ -107,97 +158,142 @@ def language(request):
...
+            exercise_file,
+            exercise / ".docs",
+        ],
+        exclude_paths=[exercise / ".docs/hints.md"],
+        no_code_map=True,
+    )
+    asyncio.ensure_future(session.start())
+    input_request_message = await session.stream.recv("input_request")
+
+    prompt_1 = dedent(
```
Comment (jakethekoenig) [inline tests/benchmarks/exercism_practice.py:231]: Thanks for moving the prompts out of the loop. That is cleaner.

**Review by jakethekoenig — APPROVED** [review]
This is awesome!

I made a few aesthetic changes to the header and made the `results.html` open automatically.

image

**Comment by jakethekoenig** [comment]
While fixing the merge conflicts I also added a `get_result_from_txt` method so if you run on some exercises and then run on a few more you can still see the full results. Probably should have done it as a separate commit but I got carried away while merging conflicts around "skip".

**Comment by jakethekoenig** [comment]
Alright sorry for pushing so much to your pull request. I'm done now. I probably should have said something when you said you were going to work on this but I didn't really expect there to be so much overlap between my refactor and automatic failure analysis.

**Comment by PCSwingle** [comment]
> Alright sorry for pushing so much to your pull request. I'm done now. I probably should have said something when you said you were going to work on this but I didn't really expect there to be so much overlap between my refactor and automatic failure analysis.

Haha, no worries, I'm just glad you handled the merge conflict because I was not looking forward to it!

File: tests/benchmarks/exercism_practice.py [hunk 2]
```diff
@@ -133,33 +98,40 @@ def language(request):
...
-    Your response will be parsed programmatically, so you MUST follow the format for the final line!
-    The possible responses for the final line and what they mean are as follows:
-    blank (the coder didn't change the file at all from the stub you provided them)
-    wording (everything was correct, but the coder messed up the wording or spacing which caused it to be rejected)
-    duplication (the coder had a random duplicated line that caused the code to not be compiled/interpreted)
-    syntax (the coder messed up their syntax, meaning their code couldn't be compiled/interpreted)
-    logic (the coder messed up the logic)
-    other (some other reason caused it to fail)""")
+prompt = (
+    "You are a professional code reviewer who helps other coders improve their skills."
```
Comment (PCSwingle) [inline tests/benchmarks/exercism_practice.py:100]: Hmm, I think we still want newlines and spaces in a lot of places here; I'm going to fix this up a bit before merging

**Pull request merged**